src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -788,11 +776,11 @@ internal Binder WithPatternVariablesIfAny(ArgumentListSyntax initializerArgument
             patterns.Free();
         }
 
-        internal BoundExpression WrapWithVariablesIfAny(BoundExpression expression)
+        internal BoundExpression WrapWithVariablesIfAny(CSharpSyntaxNode node, BoundExpression expression)
         {
             return (Locals.Length == 0)
                 ? expression
-                : new BoundSequence(expression.Syntax, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
+                : new BoundSequence(node, Locals, ImmutableArray<BoundExpression>.Empty, expression, expression.Type) { WasCompilerGenerated = true };
```

Comment:
What is the impact of `WasCompilerGenerated = true` here? Does it effect what sequence points are generated for debugging?

More broadly, `BoundSequence` has mostly been used to wrap temps while preventing side effects from being computed twice. If pattern matching never produces side effects (which I'm not sure of), do we want to consider adding a new bound node just to represent scope introduction?